src/js/builtins/ReadableStreamInternals.ts

```diff
@@ -1070,11 +1070,15 @@ export function createTextStream(highWaterMark) {
       }
 
       if (hasString && !hasBuffer) {
+        if (rope.charCodeAt(0) === 0xfeff) {
+          rope = rope.slice(1);
+        }
+
         return rope;
```

Comment:
Should this just call `withoutUTF8BOM`?

Reply:
But then the source would be readable.  This fix doesn't even have a comment explaining what it's doing.